library/src/main/java/org/apache/apex/malhar/lib/state/managed/MovingBoundaryTimeBucketAssigner.java

```diff
@@ -125,9 +125,11 @@ public long getTimeBucket(long time)
       long move = (diffInBuckets + 1) * bucketSpanMillis;
       start += move;
       end += move;
-      lowestPurgeableTimeBucket += diffInBuckets;
       // trigger purge when lower bound changes
-      triggerPurge = (diffInBuckets > 0);
+      triggerPurge = (diffInBuckets >= 0);
```

Comment:
If the value of diffInBuckets is zero, then it indicates there is a change in bucket based on below observations:
1) diffInBuckets is calculated before the end moves.
2) Tuples which are >=start and < end are valid tuples. If the time of the tuple is (=) end then time window has to be shifted.

The unit test for this change is MovingBoundaryTimeBucketAssignerTest.testTimeBucketKeyExpiry().